### PR TITLE
[e2e] Tip integration test cleanup

### PR DESCRIPTION
**Pull request by W3stside**
Cleaner approach at testing e2e for trade/fee

### Could still be separated out into chunks but this PR aims at showcasing potential new format

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr115--gpswapui.review.gnosisdev.com)**: Testing web app

File: cypress/integration/tip.test.ts [hunk 1]
```diff
@@ -1,80 +1,98 @@
...
+      // set aliases in context to reuse
+      // start fee request and alias as @fee
+      cy.request(FEE_QUERY)
+        .its('body')
+        .as('fee')
+
+      // Trade object and alias as @trade
+      cy.window()
+        .as('win')
+        .its('trade')
```
Comment (anxolin) [inline cypress/integration/tip.test.ts:46]: I don't understand how this works. Theoretically its gives you a property. Does this rely in globals? https://docs.cypress.io/api/commands/its.html#Usage

File: cypress/integration/tip.test.ts [hunk 2]
```diff
@@ -1,80 +1,98 @@
...
+        .as('trade')
+
+      cy.get('#swap-currency-output .token-amount-input').as('output')
+
+      // Clear the storage (old redux state etc)
+      cy.get<Window>('@win').then(window => window.localStorage.clear())
+    })
+
+    it('[TRADE] Trade properties are correct', () => {
```
Comment (anxolin) [inline cypress/integration/tip.test.ts:55]: I find very strange this test. It seems you want to try the SDK using integration test. 
Shouldn't we test by checking the output in the UI?

image

File: cypress/integration/tip.test.ts [hunk 3]
```diff
@@ -1,80 +1,98 @@
...
+        const { executionPrice, inputAmount } = trade
+        // OUTPUT / PRICE = INPUT (as string)
+        const calculatedOutputAmount = inputAmount.multiply(executionPrice).toSignificant(inputAmount.currency.decimals)
+        expect(calculatedOutputAmount).to.not.be.null
+        // cy.get('@output').should('equal', calculatedOutputAmount)
+      })
+    })
+
+    it('[FEE] NO FEE', () => {
+      cy.get<FeeInformation>('@fee').should(fee => {
```
Comment (anxolin) [inline cypress/integration/tip.test.ts:79]: I don't get this test either, u are calling the API and expecting to return 0 fee?
what if it changes?

**Review by anxolin — CHANGES_REQUESTED** [review]
Sorry david, I don't see the point and orientation of these tests.

I'm happy to try to sketch the tests, maybe this makes it more clear.
EDIT: This is my proposal for these tests https://github.com/gnosis/gp-swap-ui/pull/116

--- 

**More details on my comments and below 👇**

I think the aliases make it more obscure, no more clear. If we compare your tests and uniswap tests, I find their more readable because theirs are mostly self contain and small. Also they are scoped and have a clear goal

Also for context, I'll paste here my reply to your question, cause I think shows  my current thinking. 

> btw im restructuring my tests to be much cleaner... one it per section tested:
> 1. selecting DAI from list in UI
> 2. chceking Trade API exists and is filled when it should be
> 3. checking fee and blah blah

so what’s would u assert. pr what’s the goal of these ones?
I’ll try to give it a try:
1. We load the fee automatically when selecting DAI from the list. If it’s the case, I would add one extra test for ETH and xDAI. By doing so, u’’ll address this https://github.com/gnosis/gp-swap-ui/issues/110  
2. I guess you mean “Fee API”. I would assert that: `Given`: no given, `When`: we query the Fee for DAI, `Then`:  The response have the structure we expect
3. I would create different `it` for this ones:
“Trade with 0 fee”: Given no fee, `Given` DAI-WETH is selected, `When` user inputs an amount, `Then` he will receive an estimation of XX, max tokens of YYY, ..
“Trade smaller than fee”: similar to above, but the user can’t trade cause the fee is bigger than what he trades. So we assert we don’t get estimation and the trade button is disabled
“Trade small trade” similar, but we assert that if his amount is smaller than minimal fee, he pays the minimal fee
“Trade big trade” similar, but we assert that if his amount is bigger than minimal fee, he pays the fee factor value

File: cypress/integration/tip.test.ts (hunk 3 above)
Comment (alfetopito) [reply cypress/integration/tip.test.ts:79]: I have the same question, but I guess this will come later?

File: cypress/integration/tip.test.ts [hunk 4]
```diff
@@ -89,17 +107,13 @@ describe('Swap::Trade::Tip', () => {
...
-
           // input/output amounts should not be equal (ETH/DAI)
           expect(inputAmount.toExact()).to.not.equal(outputAmount.toExact())
 
-          // Subtract amountNoFee from amountWithFee to get the feeAmount
-          expect(amountWithFee.subtract(amountNoFee).toSignificant(inputAmount.currency.decimals)).to.equal(
-            feeAsCurrency?.toSignificant(inputAmount.currency.decimals) || '0'
-          )
+          const expectedFee = amountWithFee.subtract(amountNoFee).toSignificant(inputAmount.currency.decimals)
+          const calculatedFee = feeAsCurrency?.toSignificant(inputAmount.currency.decimals) || '0'
```
Comment (alfetopito) [inline cypress/integration/tip.test.ts:114]: I don't believe this optional value should be here.
The test should either expect a null value or not.
I would prefer a test where `feeAsCurrency` is null and another when it's not, to verify both paths.

**Comment by W3stside** [comment]
@alfetopito @anxolin fair enough, will re-work from #116

**Pull request closed without merging**